makeflow/src/makeflow.c

```diff
@@ -1599,6 +1626,19 @@ int main(int argc, char *argv[])
 		makeflow_gc_method = MAKEFLOW_GC_ALL;
 	}
 
+	if(wrapper_umbrella && wrapper_umbrella->spec) {
+		debug(D_MAKEFLOW_RUN, "setting dag_node->umbrella_spec...\n");
+		struct dag_node *cur;
+		cur = d->nodes;
+		while(cur) {
+			dag_node_set_umbrella_spec(cur, wrapper_umbrella->spec);
```

Comment:
@nhazekam , I got your point.
The only reason I keep it is it does not really break anything. And I indeed have plan to adding support for specifying umbrella specs for each rule.
I would keep it for now if redundancy is your only concern here.